src/github.com/couchbase/sync_gateway/db/database_test.go

```diff
@@ -742,6 +742,60 @@ func TestPostWithUserSpecialProperty(t *testing.T) {
 
 }
 
+func TestRecentSequenceHistory(t *testing.T) {
+	db := setupTestDB(t)
+	defer tearDownTestDB(t, db)
+
+	// Validate recent sequence is written
+	body := Body{"val": "one"}
+	revid, err := db.Put("doc1", body)
+
+	assert.True(t, revid != "")
+	doc, err := db.GetDoc("doc1")
+	assert.True(t, err == nil)
+	assert.DeepEquals(t, doc.RecentSequences, []uint64{1})
+
+	// Add a few revisions - validate they are retained when less than max (5)
+	for i := 0; i < 3; i++ {
+		revid, err = db.Put("doc1", body)
+	}
+
+	doc, err = db.GetDoc("doc1")
+	assert.True(t, err == nil)
+	assert.DeepEquals(t, doc.RecentSequences, []uint64{1, 2, 3, 4})
+
+	// Add many sequences to validate pruning when past max (20)
+	for i := 0; i < 20; i++ {
+		revid, err = db.Put("doc1", body)
+		// Sleep needed to ensure consistent results when running single-threaded vs. multi-threaded test:
```

Comment:
is there any way to avoid this call to sleep() in the tests?  Would it be possible to block until it came back on the tap feed?